File: packages/battery/android/src/main/java/io/capawesome/capacitorjs/plugins/battery/classes/events/BatteryStateChangeEvent.java
package io.capawesome.capacitorjs.plugins.battery.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.battery.interfaces.Result;

public class BatteryStateChangeEvent implements Result {

    @NonNull
    private final String state;

    public BatteryStateChangeEvent(@NonNull String state) {
        this.state = state;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("state", state);
        return result;
    }
}

File: packages/battery/ios/Plugin/Classes/Events/BatteryStateChangeEvent.swift
import Foundation
import Capacitor

@objc public class BatteryStateChangeEvent: NSObject, Result {
    let state: String

    init(state: String) {
        self.state = state
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["state"] = state
        return result as AnyObject
    }
}
